### PR TITLE
[BXMSDOC-4670, -4671, -4672] Review of Kogito Operator CLI messaging, CSV files, and README

### PR DESCRIPTION
**Pull request by sterobin**
See JIRAs for details:
* [BXMSDOC-4670](https://issues.jboss.org/browse/BXMSDOC-4670): General review
* [BXMSDOC-4671](https://issues.jboss.org/browse/BXMSDOC-4671): CSV files
* [BXMSDOC-4672](https://issues.jboss.org/browse/BXMSDOC-4672): CLI messaging

@ricardozanini , ready for your review. I apparently don't have perms to assign you as reviewer, so pinging you.

Just FYI, I reviewed for overall consistency, clarity, and correctness. I didn't do a full-blown review like we do with official docs, which follow strict template and stylistic requirements. But this brings things together a bit more nonetheless.

One thing, for future ref: Going forward, we should be using sentence-style headings ("Kogito CLI installation") instead of headline style headings ("Kogito CLI Installation"). I've already made this change, along with my others. This is more consistent with other Kogito docs, Red Hat docs, and the industry best practice that is becoming more prominent.

### CODE REVIEW

**Comment by ricardozanini** [comment]
@sterobin many thanks for this! We're about to merge https://github.com/kiegroup/kogito-cloud-operator/pull/104, that introduce a whole new section to the docs. Do you mind rebasing your PR after we merge it and do a quick review there also?

**Comment by sterobin** [comment]
@ricardozanini Will do.

**Comment by ricardozanini** [comment]
Hi @sterobin ! We just merged the mentioned PR, could you please rebase your PR?

Then I'll apply your changes and follow up in the future with your recommendations.

Many thanks!

**Comment by sterobin** [comment]
@ricardozanini , done. Rebased, resolved conflicts, and fixed a few other things. My changes to the TOC in the README might be hard to follow in the PR given how I had to bring in latest changes and then tweak those. So you might need to refer to the [README rendering](https://github.com/sterobin/kogito-cloud-operator/blob/BXMSDOC-4672/README.md) from my fork for the TOC at least. And I was careful and thorough with ensuring that all text and links that I changed in the TOC were updated throughout the project where applicable.

File: README.md [hunk 1]
```diff
@@ -467,131 +464,137 @@ $ kogito new-project kogito-cli
...
 $ make test
 ```
 
-## Prometheus Integration
+## Prometheus integration with the Kogito Operator
+
+### Prometheus annotations
 
-### Prometheus Annotations
+By default, if your Kogito Runtimes service contains the `monitoring-prometheus-addon` dependency, metrics for the Kogito service are enabled. For more information Prometheus metrics in Kogito services, see [Enabling metrics](https://github.com/kiegroup/kogito-runtimes/wiki/Configuration#enabling-metrics).
```
Comment (ricardozanini) [inline README.md]: For more information _about_ Prometheus metrics in Kogito services?

File: README.md [hunk 2]
```diff
@@ -2,173 +2,161 @@
...
-      * [Deploy Data Index Service](#deploy-data-index-service)
-         * [Deploy Infinispan](#deploy-infinispan)
-         * [Deploy Strimzi](#deploy-strimzi)
-            * [Create Required Topics](#create-required-topics)
-         * [Install Data Index Service](#install-data-index-service)
-            * [Retrieve Kafka internal url](#retrieve-kafka-internal-url)
-            * [Install Data Index Service with Kogito CLI](#install-data-index-service-with-kogito-cli)
-            * [Install Data Index Service with Operator Catalog (OLM)](#install-data-index-service-with-operator-catalog-olm)
-            * [Install Data Index Service with oc client](#install-data-index-service-with-oc-client)
+   * [Table of contents](#table-of-contents)
```
[the remark this answers is not in the document]
Comment (sterobin) [reply README.md]: Agreed. It was actually added in #104, here: https://github.com/kiegroup/kogito-cloud-operator/pull/104/files#diff-04c6e90faac2675aa89e2176d2eec7d8

Removed.

File: README.md (hunk 1 above)
Comment (sterobin) [reply README.md]: Oops! Fixed. Thanks.

**Comment by ricardozanini** [comment]
@xiezhang7 @radtriste could you guys please take a look?

File: README.md [hunk 3]
```diff
@@ -2,173 +2,160 @@
...
+## Kogito Operator requirements
 
-- go v1.12
-- [operator-sdk](https://github.com/operator-framework/operator-sdk/releases) v0.11.0
-- ocp 3.11/4.x (you can use [CRC](https://github.com/code-ready/crc) for local deployment)
-- [kogito s2i imagestreams](https://raw.githubusercontent.com/kiegroup/kogito-cloud/master/s2i/kogito-imagestream.yaml) installed
+- Go v1.12 is installed.
+- The [operator-sdk](https://github.com/operator-framework/operator-sdk/releases) v0.11.0 is installed.
+- OpenShift 3.11 or 4.x is installed. (You can use [CRC](https://github.com/code-ready/crc) for local deployment.)
+- The [kogito s2i imagestreams](https://raw.githubusercontent.com/kiegroup/kogito-cloud/master/s2i/kogito-imagestream.yaml) is installed.
```
[the remark this answers is not in the document]
Comment (radtriste) [reply README.md]: @ricardozanini This is not a requirement or ? Just after, we put it as "Optional" in the document beacuse Operator does install them ...

File: README.md [hunk 4]
```diff
@@ -2,173 +2,160 @@
...
-### Deploy to OpenShift 4.x
+### Deploying to OpenShift 4.x
 
-Kogito operator is a namespaced operator, which means that you will need to install it into the namespace you want your Kogito application to run in.
+The Kogito operator is a namespaced operator, so you must install it into the namespace where you want your Kogito application to run.
 
-#### Optional Step
-
-You can import the Kogito image stream using the `oc client` manually with the following command:
+(Optional) You can import the Kogito image stream using the `oc client` manually with the following command:
```
Comment (radtriste) [inline README.md]: @ricardozanini Here we put it as optional

File: README.md [hunk 5]
```diff
@@ -2,173 +2,160 @@
...
 
 ```bash
 $ oc apply -f https://raw.githubusercontent.com/kiegroup/kogito-cloud/master/s2i/kogito-imagestream.yaml -n openshift
 ```
 
-But, this step is not mandatory anymore, when kogito is going to install a new app, it will create the required imagestreams
-before.
+This step is optional because Kogito creates the required imagestreams when it installs a new application.
```
Comment (radtriste) [inline README.md]: "because the Kogito Operator creates" ?

File: README.md [hunk 6]
```diff
@@ -2,173 +2,160 @@
...
 
-Just follow the OpenShift Web Console instructions in the _Catalog_, _OperatorHub_ section in the left menu to install it in any namespace in the cluster. 
+You can also verify the Operator availability in the catalog by running the following command:
+
+```bash
+$ oc describe operatorsource.operators.coreos.com/kogitocloud-operator -n openshift-marketplace
+```
+Follow the OpenShift Web Console instructions in the **Catalog** -> **OperatorHub** section in the left menu to install it in any namespace in the cluster.
 
 ![Kogito Operator in the Catalog](docs/img/catalog-kogito-operator.png?raw=true)
```
Comment (radtriste) [inline README.md]: Should we update the screenshot ? Or wait for version 1.0 ?

File: README.md [hunk 7]
```diff
@@ -177,17 +164,17 @@ NAME                                     READY   STATUS      RESTARTS   AGE
...
 
 ```bash
 $ oc logs -f kogito-cloud-operator-6d7b6d4466-9ng8t
 ```
 
-#### Native build fails with "The build pod was killed due to an out of memory condition."
+#### Native build fails with "The build pod was killed due to an out of memory condition." error
 
-By default, the operator will set the limit in the s2i build pod to 10GB of memory and 1 CPU. It might happen that your Kogito service would require even more resources from the cluster. 
+By default, the Operator sets the limit in the S2I build pod to 10GB of memory and 1 CPU. Your Kogito service might require more resources from the cluster.
```
Comment (radtriste) [inline README.md]: @ricardozanini This should be deleted has we don't have build limits anymore in 0.6.0 ?

File: README.md [hunk 8]
```diff
@@ -204,46 +191,48 @@ spec:
...
-## Deploy Data Index Service
+The Kogito Operator can deploy the [Data Index Service](https://github.com/kiegroup/kogito-runtimes/wiki/Data-Index-Service) as a [Custom Resource](deploy/crds/app.kiegroup.org_v1alpha1_kogitodataindex_cr.yaml) (`KogitoDataIndex`).
 
-The Kogito Operator is able to deploy the [Data Index Service](https://github.com/kiegroup/kogito-runtimes/wiki/Data-Index-Service) as a [Custom Resource](deploy/crds/app.kiegroup.org_v1alpha1_kogitodataindex_cr.yaml) (`KogitoDataIndex`). 
+The Data Index Service depends on Kafka, so you must manually deploy an Apache Kafka Cluster, such as [Strimzi](https://strimzi.io/).
 
-Since Data Index Service depends on Kafka, it's necessary to manually deploy an Apache Kafka Cluster (we use [Strimzi](https://strimzi.io/)) in the same namespace.
- 
-Data Index also depends on Infinispan, but since 0.6.0 version, Infinispan Server is *automagically* deployed for you.
+The Data Index Service also depends on Infinispan, but starting in version 0.6.0 of the Kogito Operator, Infinispan Server is automatically deployed for you.
```
Comment (radtriste) [inline README.md]: "starting **with** version 0.6.0" ?

File: README.md [hunk 9]
```diff
@@ -204,46 +191,48 @@ spec:
...
 | --- |
 
-### Deploy Infinispan
+### Deploying Infinispan
 
-If you plan to have the Data Index to connect to the Infinispan server deployed within the same namespace, Kogito Operator can handle this deployment for you.
+If you plan to use the Data Index Service to connect to an Infinispan Server instance deployed within the same namespace, the Kogito Operator can handle this deployment for you.
 
-When installing the Kogito Operator from OperatorHub, the Infinispan Operator will be available in the same namespace. Otherwise, if you don't have OperatorHub or OLM available in your cluster, you'll have to [subscribe to Infinispan Operator](https://infinispan.org/infinispan-operator/master/documentation/asciidoc/titles/operator.html#deploying_operator_manually) manually.
+When you install the Kogito Operator from OperatorHub, the Infinispan Operator is available in the same namespace. If you do not have access to OperatorHub or OLM in your cluster, you can [manually deploy the Infinispan Operator](https://infinispan.org/infinispan-operator/master/documentation/asciidoc/titles/operator.html#deploying_operator_manually).
```
Comment (radtriste) [inline README.md]: When you install the Kogito Operator from OperatorHub, the Infinispan Operator should be installed in the same namespace when first installing the Data Index Service.
@ricardozanini for confirmation

File: README.md [hunk 10]
```diff
@@ -204,46 +191,48 @@ spec:
...
 
-Having the Infinispan Operator deployed, just jump to [Install Data Index Service section](#install-data-index-service).
+After you deploy the Infinispan Operator, see [Kogito Data Index Service installation](#kogito-data-index-service-installation) for next steps.
```
Comment (radtriste) [inline README.md]: That link just skips the whole Strimzi deployment ... Is that on purpose ?

File: README.md [hunk 11]
```diff
@@ -255,20 +244,20 @@ my-cluster-zookeeper-client   ClusterIP   172.30.241.146   <none>        2181/TC
...
+After the cluster is running, create the following Kafka Topics that are required by the Data Index Service:
 
 - `kogito-processinstances-events`
-- `kogito-usertaskinstances-events` 
+- `kogito-usertaskinstances-events`
 - `kogito-processdomain-events`
 - `kogito-usertaskdomain-events`
 
-To create the required topics, in the OpenShift Web Console go to the Installed Operators, Strimzi Operator, Kafka Topic tab. 
+For each required Kafka Topic, in the OpenShift Web Console go to **Installed Operators** -> **Strimzi Operator** -> **Kafka Topic** and create the Topic.
```
Comment (radtriste) [inline README.md]: We have to put a note here that:

If this is a development environment, and the user set the Zookeeper and Kafka replicas to 1 to conserve resources, then it should set also the replicas to 1 in each topic (I got errors on topics by letting the '3' replicas)

File: README.md [hunk 12]
```diff
@@ -285,9 +274,8 @@ spec:
     retention.ms: 604800000
     segment.bytes: 1073741824
 ```
-And then do the same for the others topics listed above.
 
-To check if everything was created successfully run the following command:
+After you create all required Kafka Topics, run the following command to verify that the Topics were created successfully:
 
 ```bash
 $ oc describe kafkatopic/kogito-processinstances-events
```
Comment (radtriste) [inline README.md]: Here we can see the number of replica to 1 :)

File: README.md [hunk 13]
```diff
@@ -315,124 +303,132 @@ Spec:
...
 
-### CLI Requirements
+### Kogito CLI requirements
 
-1. [`oc` client](https://docs.okd.io/latest/cli_reference/get_started_cli.html) installed
-2. An authenticated OpenShift user with permissions to create resources in a given namespace
-3. Kogito Operator [installed](#installation) in the cluster
+- The [`oc` client](https://docs.okd.io/latest/cli_reference/get_started_cli.html) is installed.
+- You are an authenticated OpenShift user with permissions to create resources in a given namespace.
+- The [Kogito Operator](#kogito-operator-installation) is installed in the cluster.
```
Comment (radtriste) [inline README.md]: I don't see it as a requirement as the CLI tries it install it on project where it is not (or at least the CLI will ask to install it via OperatorHub).

File: docs/ARCHITECTURE.MD [hunk 14]
```diff
@@ -1,22 +1,26 @@
-# Architecture
+# Kogito Operator architecture
 
-The actual architecture has two [controllers](https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg#hdr-Controller) that's responsible for deploying the application from source and to deploy the [Data Index Service](https://github.com/kiegroup/kogito-runtimes/wiki/Data-Index-Service). It's on the [roadmap](https://github.com/kiegroup/kogito-runtimes/wiki/Roadmap) to have one more controller to handle SSO. The following image illustrates the general idea:
+The Kogito Operator architecture uses two [controllers](https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg#hdr-Controller) to deploy the application from source and to deploy the [Data Index Service](https://github.com/kiegroup/kogito-runtimes/wiki/Data-Index-Service). One more controller to handle SSO is on the [Kogito roadmap](https://github.com/kiegroup/kogito-runtimes/wiki/Roadmap).
```
[the remark this answers is not in the document]
Comment (ricardozanini) [reply docs/ARCHITECTURE.MD]: I’d say that’s Kogito Data Index Service, but I think @sterobin has a justification for this naming change based on the docs requirements perhaps?

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (ricardozanini) [reply README.md]: I’d say to change this to optional since we are installing the images automatically. But should be changed on the JIRA that we introduced this feature. We made a mistake. @sterobin fo you mind adding an “optional” note on this? Thanks!

File: README.md (hunk 6 above)
Comment (ricardozanini) [reply README.md]: let’s wait for 1.0

File: README.md (hunk 9 above)
Comment (ricardozanini) [reply README.md]: Actually when installing the Operator from Hub, the OLM will install the Infinispan operator for us since it’s a requirement in the CSV file. So I’d change to “will install” instead of “should”. @sterobin wdyt?

File: docs/ARCHITECTURE.MD (hunk 14 above)
[the remark this answers is not in the document]
Comment (sterobin) [reply docs/ARCHITECTURE.MD]: Done

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (sterobin) [reply README.md]: Sure, but I just removed. The section is about "requirements", so we can't have an optional requirement. Plus it's mentioned in the subsequent section where relevant, as you said.

File: README.md (hunk 5 above)
Comment (sterobin) [reply README.md]: Done

File: README.md (hunk 7 above)
Comment (sterobin) [reply README.md]: @ricardozanini , please confirm.

File: README.md (hunk 8 above)
Comment (sterobin) [reply README.md]: Sure, done.

File: README.md (hunk 9 above)
Comment (sterobin) [reply README.md]: Will use "is" though, because best practice to avoid future tense as much as possible. But @radtriste @ricardozanini, is Infinispan installed when the Operator is installed or when the Data Index Service is installed? @radtriste wording basically says both, which confuses me.

I've written just as this for now:

>When you install the Kogito Operator from OperatorHub, the Infinispan Operator is installed in the same namespace.

File: README.md (hunk 10 above)
Comment (sterobin) [reply README.md]: @ricardozanini , since that's how it already was, please confirm.

File: README.md (hunk 11 above)
Comment (sterobin) [reply README.md]: Done. Added the following note:

```
| :information_source: If you are using a development environment and you set the Zookeeper and Kafka replicas to `1` to conserve resources, ensure that you also set the replicas to `1` in each Kafka Topic. The replica setting must match or you might encounter errors with the Kafka Topics. |
| --- |
```

File: README.md (hunk 13 above)
Comment (sterobin) [reply README.md]: Removed

File: README.md (hunk 9 above)
Comment (ricardozanini) [reply README.md]: @sterobin when Kogito Operator is installed via OperatorHub, Infinispan Operator is also installed. When Data Index is deployed with `UseKogitoInfra` parameter set to `true`, Infinispan Server is deployed to serve as the persistence actor in the infrastructure.

File: README.md (hunk 10 above)
Comment (ricardozanini) [reply README.md]: It's a mistake, good point @radtriste. This should be changed to "see Deploy Strimzi" for next steps. Good catch!

File: README.md (hunk 7 above)
Comment (ricardozanini) [reply README.md]: @radtriste correct

File: README.md (hunk 9 above)
Comment (ricardozanini) [reply README.md]: @sterobin do you guys have a document for writing best practices? Even though I know I lack English knowledge, will be nice to have something like this to have in mind when writing READMEs.

File: README.md (hunk 7 above)
Comment (sterobin) [reply README.md]: Deleted that section on builds failing due to insufficient build limit.

File: README.md (hunk 9 above)
Comment (sterobin) [reply README.md]: I've added what you've suggested and rearranged this bit as follows. If this is still not right or not what you wanted, please correct it directly so that it's accurate:

> When you install the Kogito Operator from OperatorHub, the Infinispan Operator is installed in the same namespace. When you install the Data Index Service with the `UseKogitoInfra` parameter set to `true`, Infinispan Server is deployed to serve as the persistence actor in the infrastructure. 

> If you do not have access to OperatorHub or OLM in your cluster, you can [manually deploy the Infinispan Operator](https://infinispan.org/infinispan-operator/master/documentation/asciidoc/titles/operator.html#deploying_operator_manually).
```

@ricardozanini , regarding style, Red Hat technical documentation standards are based on the [IBM Style Guide](https://www.amazon.com/IBM-Style-Guide-Conventions-Writers/dp/0132101300/ref=sr_1_1?keywords=IBM+style+guide&qid=1574270099&s=books&sr=1-1) (coincidentally), of which there's an abbreviated [version online for developers](https://www.ibm.com/developerworks/library/styleguidelines/index.html). There are various tech comm style guides that I've had to ingest (Microsoft, Sun Microsystems, IBM, etc.) but for your needs, that online IBM version should be sufficient. 

I also strongly recommend to anyone I talk to about this to look into the [Global English Style Guide](https://www.amazon.com/Global-English-Style-Guide-Documentation/dp/1599946572/ref=sr_1_1?keywords=global+english+style+guide&qid=1574270270&s=books&sr=1-1). An invaluable asset for mass communicators, particularly in open source communities. I became good friends with the author, now retired from SAS.

File: README.md (hunk 10 above)
Comment (sterobin) [reply README.md]: Done.

File: README.md (hunk 9 above)
Comment (ricardozanini) [reply README.md]: Gosh! How many valuable information there! Many thanks for this, @sterobin

File: README.md (hunk 7 above)
Comment (radtriste) [reply README.md]: Don't forget also to update the TOC :)

File: README.md [hunk 15]
```diff
@@ -2,173 +2,159 @@
...
+            * [Manually in OperatorHub](#manually-in-operatorhub)
+            * [Locally on your system](#locally-on-your-system)
+         * [Deploying to OpenShift 3.11](#deploying-to-openshift-311)
+      * [Kogito Runtimes service deployment](#kogito-runtimes-service-deployment)
+         * [Deploying a new service](#deploying-a-new-service)
+         * [Cleaning up a Kogito service deployment](#cleaning-up-a-kogito-service-deployment)
+         * [Native X JVM builds](#native-x-jvm-builds)
+         * [Troubleshooting Kogito Runtimes service deployment](#troubleshooting-kogito-runtimes-service-deployment)
+            * [No builds are running](#no-builds-are-running)
+            * [Native build fails with "The build pod was killed due to an out of memory condition." error](#native-build-fails-with-the-build-pod-was-killed-due-to-an-out-of-memory-condition-error)
```
Comment (radtriste) [inline README.md]: Don't forget also to update the TOC :)

Comment (sterobin) [reply README.md]: Oops! Sorry. Good catch. Fixed. Thank you!

**Comment by sterobin** [comment]
@ricardozanini , all fixed, and re-squashed.
